ada/extensions/src/libadalang-doc_utils.adb

```diff
@@ -136,12 +136,29 @@ package body Libadalang.Doc_Utils is
       --  Transform the doc vector into a string
       declare
          Last_Index : Natural := Doc_Vec.Last_Index;
+
+         First_Line : XString :=
+           (if Last_Index /= 0 then Doc_Vec.First_Element else Null_XString);
+
+         Offset     : Positive :=
+           First_Line.Length - First_Line.Trim.Length + 1;
+         --  Offset for the first non whitespace char on the first line. Will
+         --  be used as the offset to start every subsequent doc line.
+
       begin
-         for I in Doc_Vec.First_Index .. Doc_Vec.Last_Index loop
+         for I in Doc_Vec.First_Index .. Last_Index loop
             declare
                L : XString renames Doc_Vec (I);
             begin
-               Ret.Doc.Append (L.Slice (3, L.Length));
+
+               --  Check that every character we're going to strip is a white
+               --  space; else, raise an error.
+               if not (for all C of L.Slice (1, Offset - 1) => Is_Space (C))
+               then
+                  raise Property_Error with "Invalid formatted docstring";
```

Comment:
I believe “docstring” designates string literals (and will thus be surprising for a lot of Ada programmers): here we’re just processing comments, so I think we should reword to something like “Invalid indentation in documentation comment”.

Reply:
Good point